File: project/app/src/main/java/org/owntracks/android/support/widgets/EditIntegerPreference.java
package org.owntracks.android.support.widgets;

import android.content.Context;

import android.util.AttributeSet;

import timber.log.Timber;

public class EditIntegerPreference extends EditStringPreference {

    public EditIntegerPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public EditIntegerPreference(Context context) {
        super(context);
    }

    @Override
    protected boolean persistString(String value) {
        if(value == null || "".equals(value)) {
            getSharedPreferences().edit().remove(getKey()).apply();
            return true;
        }
        try {
            return persistInt(Integer.valueOf(value));
        } catch (NumberFormatException e) {
            return false;
        }
    }

    @Override
    protected String getPersistedString(String defaultReturnValue) {
        if(getSharedPreferences().contains(getKey())) {
            try {
                int intValue = getPersistedInt(0);
                return String.valueOf(intValue);
            }  catch (ClassCastException e) {
                Timber.e("Error retriving string preference %s, returning default", defaultReturnValue);
                return defaultReturnValue;
            }
        } else {
            return defaultReturnValue;
        }
    }
}

File: project/app/src/main/java/org/owntracks/android/support/widgets/EditIntegerPreference.kt
package org.owntracks.android.support.widgets

import android.content.Context
import android.util.AttributeSet
import androidx.preference.ValidatingEditTextPreference
import timber.log.Timber

class EditIntegerPreference(context: Context, attrs: AttributeSet?) :
    ValidatingEditTextPreference(context, attrs) {
  override fun persistString(value: String?): Boolean {
    return try {
      persistInt(Integer.valueOf(value ?: "0"))
    } catch (e: NumberFormatException) {
      false
    }
  }

  override fun getPersistedString(defaultReturnValue: String?): String? {
    return try {
      getPersistedInt(0).toString()
    } catch (e: ClassCastException) {
      Timber.e("Error retrieving string preference $key, returning default")
      defaultReturnValue
    }
  }
}
